### DISCUSSION TITLE
Remove `Type.filter_inplace`

### DISCUSSION THREAD

**Discussion by markusschmaus** [aesara-devs/aesara#1248] [General]
`Type.filter_inplace` is currently not implemented anywhere in the code and the only place where it is used the `NotImplementedError` is caught and `Type.filter` is used instead. Based on the comment it seems that this is some leftover from some earlier feature. Is there any reason why this should be kept around? Otherwise I am going to remove it in https://github.com/aesara-devs/aesara/pull/1207.

**Comment by brandonwillard** [comment]
Yeah, there's still a bit of left over machinery from Theano, and that's a good example.
